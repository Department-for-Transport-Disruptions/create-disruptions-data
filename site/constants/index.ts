import {
    MiscellaneousReason,
    PersonnelReason,
    EnvironmentReason,
    EquipmentReason,
} from "@create-disruptions-data/shared-ts/enums";

export const SUPPORT_EMAIL_ADDRESS = "bodshelpdesk@kpmg.co.uk";

export const SERVICE_EMAIL_ADDRESS = "bodshelpdesk@kpmg.co.uk";

export const SUPPORT_PHONE_NUMBER = "0800 123 1234";

export const DISRUPTION_REASONS: {
    value: MiscellaneousReason | PersonnelReason | EnvironmentReason | EquipmentReason;
    display: string;
}[] = [
    {
        value: MiscellaneousReason.roadWorks,
        display: "Road Works",
    },
    {
        value: MiscellaneousReason.vandalism,
        display: "Vandalism",
    },
    {
        value: MiscellaneousReason.routeDiversion,
        display: "Route Diversion",
    },
    {
        value: MiscellaneousReason.specialEvent,
        display: "Special Event",
    },
];

export const DISRUPTION_TYPES = ["planned", "unplanned"];

export const CD_DATE_FORMAT = "DD/MM/YYYY";
export const CONTACT_FEEDBACK_QUESTION = "Did you contact us for assistance at any point?";
export const SOLVE_FEEDBACK_QUESTION = "Did we solve your problem?";
export const HEAR_ABOUT_US_FEEDBACK_QUESTION = "How did you hear about our service?";
export const GENERAL_FEEDBACK_QUESTION = "Please let us know any feedback or suggestions for improvements you may have";

export const COOKIES_POLICY_COOKIE = "cdd-cookies-policy";

export const COOKIE_PREFERENCES_COOKIE = "cdd-cookie-preferences-set";

export const oneYearInSeconds = 31556952;

export const ID_TOKEN_COOKIE = "cdd-id-token";

export const COOKIES_DISRUPTION_INFO = "disruption-info";

export const COOKIES_DISRUPTION_ERRORS = "disruption-errors";

export const COOKIES_ADD_CONSEQUENCE_INFO = "cdd-consequence-info";

export const COOKIES_ADD_CONSEQUENCE_ERRORS = "cdd-consequence-errors";

export const CREATE_DISRUPTION_PAGE_PATH = "/create-disruption";

export const ERROR_PATH = "/_error";

export const ADD_CONSEQUENCE_PAGE_PATH = "/type-of-consequence";

export const STAGE = process.env.STAGE || "dev";
