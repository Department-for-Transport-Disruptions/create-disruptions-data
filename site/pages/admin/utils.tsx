import resendInvite from "../api/admin/resend-invite.api";

const createUserActionsLink = (
    key: string,
    index: number,
    username: string,
    userGroup: string,
    showResendInvite?: boolean,
) => {
    return (
        <>
            {showResendInvite ? (
                <>
                    <button
                        key={`${key}${index ? `-${index}` : ""}`}
                        className="govuk-link"
                        onClick={() => resendInvite(username, userGroup)}
                    >
                        Resend invite
                    </button>
                    <br />
                    <button
                        key={`${key}${index ? `-remove-${index}` : "-remove"}`}
                        className="govuk-link"
                        onClick={() => removeUser(username)}
                    >
                        Remove
                    </button>
                </>
            ) : (
                <button
                    key={`${key}${index ? `-${index}` : ""}`}
                    className="govuk-link"
                    onClick={() => removeUser(username)}
                >
                    Remove
                </button>
            )}
        </>
    );
};
