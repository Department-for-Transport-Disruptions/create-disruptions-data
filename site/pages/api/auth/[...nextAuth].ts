import NextAuth from "next-auth";
import CognitoProvider from "next-auth/providers/cognito";

export default NextAuth({
    providers: [
        CognitoProvider({
            clientId: "",
            clientSecret: "",
            issuer: "",
        }),
    ],
    pages: {
        signIn: "",
        newUser: "",
    },
    session: {
        strategy: "jwt",
    },
});
