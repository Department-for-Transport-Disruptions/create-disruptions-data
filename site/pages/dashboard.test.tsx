import { getDate } from "@create-disruptions-data/shared-ts/utils/dates";
import renderer from "react-test-renderer";
import { describe, it, expect, vi, afterEach, beforeEach } from "vitest";
import { randomUUID } from "crypto";
import Dashboard, { DashboardDisruption, getServerSideProps } from "./dashboard.page";
import { CD_DATE_FORMAT } from "../constants";
import * as dynamo from "../data/dynamo";
import { defaultModes } from "../schemas/organisation.schema";
import { SessionWithOrgDetail } from "../schemas/session.schema";
import { disruptionArray, disruptionWithConsequencesAndSocialMediaPosts, getMockContext } from "../testData/mockData";
import * as session from "../utils/apiUtils/auth";

const getDisruptionsSpy = vi.spyOn(dynamo, "getPublishedDisruptionsDataFromDynamo");
const getPendingDisruptionsSpy = vi.spyOn(dynamo, "getPendingDisruptionsIdsFromDynamo");
vi.mock("../data/dynamo");

const getSessionWithOrgDetailSpy = vi.spyOn(session, "getSessionWithOrgDetail");
vi.mock("../utils/apiUtils/auth", async () => ({
    ...(await vi.importActual<object>("../utils/apiUtils/auth")),
    getSession: vi.fn(),
}));

beforeEach(() => {
    getSessionWithOrgDetailSpy.mockResolvedValue(defaultSession);
});
const defaultNewDisruptionId = "acde070d-8c4c-4f0d-9d8a-162843c10333";

const recentlyClosedDate = getDate().subtract(4, "day").format(CD_DATE_FORMAT);
const isoRecentlyClosedDate = getDate().subtract(4, "day").format("YYYY-MM-DD");

const defaultSession: SessionWithOrgDetail = {
    email: "test@example.com",
    isOrgAdmin: false,
    isOrgPublisher: false,
    isOrgStaff: false,
    isSystemAdmin: true,
    orgId: randomUUID(),
    username: "test@example.com",
    name: "Test User",
    orgName: "Nexus",
    adminAreaCodes: ["A", "B", "C"],
    mode: defaultModes,
};

const disruptions: DashboardDisruption[] = [
    {
        id: "12",
        summary: "A bad disruption",
        validityPeriods: [
            {
                startTime: "2023-03-21T11:23:24.529Z",
                endTime: null,
            },
        ],
        displayId: "8fg3ha",
    },
    {
        id: "33",
        summary: "A more ok disruption",
        validityPeriods: [
            {
                startTime: "2023-03-21T11:23:24.529Z",
                endTime: "2023-03-22T11:23:24.529Z",
            },
        ],
        displayId: "8fg3ha",
    },
    {
        id: "44",
        summary: "Another disruption",
        validityPeriods: [
            {
                startTime: "2023-04-21T11:23:24.529Z",
                endTime: "2024-03-22T11:23:24.529Z",
            },
            {
                startTime: "2023-04-22T11:23:24.529Z",
                endTime: null,
            },
        ],
        displayId: "8fg3ha",
    },
    {
        id: "55",
        summary: "Another disruption",
        validityPeriods: [
            {
                startTime: "2023-04-21T11:23:24.529Z",
                endTime: "2024-03-22T11:23:24.529Z",
            },
            {
                startTime: "2023-04-22T11:23:24.529Z",
                endTime: null,
            },
        ],
        displayId: "8fg3ha",
    },
];

describe("pages", () => {
    describe("dashboard", () => {
        it("should render correctly when there are no disruptions", () => {
            const tree = renderer
                .create(
                    <Dashboard
                        liveDisruptions={[]}
                        upcomingDisruptions={[]}
                        recentlyClosedDisruptions={[]}
                        newDisruptionId={defaultNewDisruptionId}
                        canPublish
                        orgName="Nexus"
                        stage="dev"
                    />,
                )
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        it("should render correctly when there are only live disruptions", () => {
            const tree = renderer
                .create(
                    <Dashboard
                        liveDisruptions={disruptions}
                        upcomingDisruptions={[]}
                        recentlyClosedDisruptions={[]}
                        newDisruptionId={defaultNewDisruptionId}
                        canPublish
                        orgName="Nexus"
                        stage="dev"
                    />,
                )
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        it("should render correctly when there are only upcoming disruptions", () => {
            const tree = renderer
                .create(
                    <Dashboard
                        liveDisruptions={[]}
                        upcomingDisruptions={disruptions}
                        recentlyClosedDisruptions={[]}
                        newDisruptionId={defaultNewDisruptionId}
                        canPublish
                        orgName="Nexus"
                        stage="dev"
                    />,
                )
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        it("should render correctly when there are only recently closed disruptions", () => {
            const tree = renderer
                .create(
                    <Dashboard
                        liveDisruptions={[]}
                        upcomingDisruptions={[]}
                        recentlyClosedDisruptions={disruptions}
                        newDisruptionId={defaultNewDisruptionId}
                        canPublish
                        orgName="Nexus"
                        stage="dev"
                    />,
                )
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        it("should render correctly when there are all three live, upcoming and recently closed disruptions", () => {
            const tree = renderer
                .create(
                    <Dashboard
                        liveDisruptions={[disruptions[0]]}
                        upcomingDisruptions={[disruptions[1], disruptions[2]]}
                        recentlyClosedDisruptions={[disruptions[3]]}
                        newDisruptionId={defaultNewDisruptionId}
                        canPublish
                        orgName="Nexus"
                        stage="dev"
                    />,
                )
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        describe("getServerSideProps", () => {
            afterEach(() => {
                vi.resetAllMocks();
            });

            beforeEach(() => {
                getPendingDisruptionsSpy.mockResolvedValue(new Set<string>());
            });

            it("should return no disruptions if there is no data returned from the database call", async () => {
                getDisruptionsSpy.mockResolvedValue([]);

                const ctx = getMockContext();

                const actualProps = await getServerSideProps(ctx);
                expect(actualProps.props).toStrictEqual({
                    liveDisruptions: [],
                    upcomingDisruptions: [],
                    recentlyClosedDisruptions: [],
                    newDisruptionId: expect.any(String) as string,
                    pendingApprovalCount: 0,
                    canPublish: true,
                    orgName: "Nexus",
                    
                });
            });

            it("should return live disruptions if the data returned from the database has live dates", async () => {
                getDisruptionsSpy.mockResolvedValue([disruptionWithConsequencesAndSocialMediaPosts]);
                const ctx = getMockContext();

                const actualProps = await getServerSideProps(ctx);
                expect(actualProps.props).toStrictEqual({
                    liveDisruptions: [
                        {
                            id: "acde070d-8c4c-4f0d-9d8a-162843c10333",
                            summary: "Some summary",
                            validityPeriods: [
                                {
                                    startTime: "2023-03-10T12:00:00.000Z",
                                    endTime: null,
                                },
                                { startTime: "2023-03-18T12:00:00.000Z", endTime: null },
                            ],
                            displayId: "8fg3ha",
                        },
                    ],
                    upcomingDisruptions: [],
                    recentlyClosedDisruptions: [],
                    newDisruptionId: expect.any(String) as string,
                    pendingApprovalCount: 0,
                    canPublish: true,
                    orgName: "Nexus",
                    
                });
            });

            it("should return upcoming disruptions if the data returned from the database has upcoming dates", async () => {
                getDisruptionsSpy.mockResolvedValue([
                    {
                        ...disruptionWithConsequencesAndSocialMediaPosts,
                        validity: [],
                        disruptionStartDate: "12/02/2999",
                        disruptionStartTime: "1300",
                    },
                ]);

                const ctx = getMockContext();

                const actualProps = await getServerSideProps(ctx);

                expect(actualProps.props).toStrictEqual({
                    liveDisruptions: [],
                    upcomingDisruptions: [
                        {
                            id: "acde070d-8c4c-4f0d-9d8a-162843c10333",
                            summary: "Some summary",
                            validityPeriods: [{ startTime: "2999-02-12T13:00:00.000Z", endTime: null }],
                            displayId: "8fg3ha",
                        },
                    ],
                    recentlyClosedDisruptions: [],
                    newDisruptionId: expect.any(String) as string,
                    pendingApprovalCount: 0,
                    canPublish: true,
                    orgName: "Nexus",
                    
                });
            });

            it("should return recently closed disruptions if the data returned from the database has end dates within 7 days", async () => {
                getDisruptionsSpy.mockResolvedValue([
                    {
                        ...disruptionWithConsequencesAndSocialMediaPosts,
                        validity: [],
                        disruptionStartDate: "12/02/2023",
                        disruptionStartTime: "1300",
                        disruptionEndDate: recentlyClosedDate,
                        disruptionEndTime: "1300",
                        disruptionNoEndDateTime: undefined,
                    },
                ]);

                const ctx = getMockContext();

                const actualProps = await getServerSideProps(ctx);

                expect(actualProps.props).toStrictEqual({
                    liveDisruptions: [],
                    recentlyClosedDisruptions: [
                        {
                            id: "acde070d-8c4c-4f0d-9d8a-162843c10333",
                            summary: "Some summary",
                            validityPeriods: [
                                {
                                    startTime: "2023-02-12T13:00:00.000Z",
                                    endTime: `${isoRecentlyClosedDate}T12:00:00.000Z`,
                                },
                            ],
                            displayId: "8fg3ha",
                        },
                    ],
                    upcomingDisruptions: [],
                    newDisruptionId: expect.any(String) as string,
                    pendingApprovalCount: 0,
                    canPublish: true,
                    orgName: "Nexus",
                    
                });
            });

            it("should return live and upcoming disruptions if the data returned from the database has live and upcoming dates", async () => {
                getDisruptionsSpy.mockResolvedValue([
                    ...disruptionArray,
                    {
                        ...disruptionWithConsequencesAndSocialMediaPosts,
                        validity: [],
                        disruptionStartDate: "12/02/2999",
                        disruptionStartTime: "1200",
                    },
                ]);

                const ctx = getMockContext();

                const actualProps = await getServerSideProps(ctx);

                expect(actualProps.props).toStrictEqual({
                    liveDisruptions: [
                        {
                            id: "acde070d-8c4c-4f0d-9d8a-162843c10333",
                            summary: "Some summary",
                            validityPeriods: [{ startTime: "2022-03-10T11:00:00.000Z", endTime: null }],
                            displayId: "8fg3ha",
                        },
                        {
                            id: "acde070d-8c4c-4f0d-9d8a-162843c10333",
                            summary: "Some summary",
                            validityPeriods: [
                                {
                                    startTime: "2023-03-10T12:00:00.000Z",
                                    endTime: null,
                                },
                                { startTime: "2023-03-18T12:00:00.000Z", endTime: null },
                            ],
                            displayId: "8fg3ha",
                        },
                        {
                            id: "acde070d-8c4c-4f0d-9d8a-162843c10333",
                            summary: "Some summary",
                            validityPeriods: [
                                {
                                    startTime: "2023-03-10T12:00:00.000Z",
                                    endTime: null,
                                },
                                { startTime: "2023-03-18T12:00:00.000Z", endTime: null },
                            ],
                            displayId: "8fg3ha",
                        },
                    ],
                    upcomingDisruptions: [
                        {
                            id: "acde070d-8c4c-4f0d-9d8a-162843c10333",
                            summary: "Some summary",
                            validityPeriods: [{ startTime: "2999-02-12T12:00:00.000Z", endTime: null }],
                            displayId: "8fg3ha",
                        },
                    ],
                    recentlyClosedDisruptions: [],
                    newDisruptionId: expect.any(String) as string,
                    pendingApprovalCount: 0,
                    canPublish: true,
                    orgName: "Nexus",
                    
                });
            });
        });
    });
});
