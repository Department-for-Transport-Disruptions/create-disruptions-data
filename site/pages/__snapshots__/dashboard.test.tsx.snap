// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`pages > dashboard > should render correctly 1`] = `
<DocumentFragment>
  <div
    class="app-phase-banner__wrapper"
  >
    <div
      class="govuk-phase-banner max-w-[960px] w-[90%] border-0 mx-auto app-phase-banner app-width-container"
    >
      <p
        class="govuk-phase-banner__content"
      >
        <strong
          class="govuk-tag govuk-phase-banner__content__tag"
        >
          beta
        </strong>
        <span
          class="govuk-phase-banner__text"
        >
          This is a new service â€“ your 
          <a
            class="govuk-link"
            href="/feedback"
            id="feedback-link"
          >
            feedback
          </a>
           will help us to improve it.
        </span>
      </p>
    </div>
  </div>
  <div
    class="govuk-width-container"
  >
    <main
      class="govuk-main-wrapper"
      id="main-content"
    >
      <h1
        class="govuk-heading-xl"
      >
        Test Org disruptions data
      </h1>
      <a
        class="govuk-button govuk-button--start"
        data-module="govuk-button"
        draggable="false"
        href="/create-disruption/acde070d-8c4c-4f0d-9d8a-162843c10333"
        id="create-new-button"
        role="button"
      >
        Create new disruption
        <svg
          class="govuk-button__start-icon"
          focusable="false"
          height="19"
          role="presentation"
          viewBox="0 0 33 40"
          width="17.5"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h13l20 20-20 20H0l20-20z"
            fill="currentColor"
          />
        </svg>
      </a>
      <div
        class="govuk-tabs"
        data-module="govuk-tabs"
      >
        <h2
          class="govuk-tabs__title"
        >
          Disruptions
        </h2>
        <ul
          class="govuk-tabs__list"
        >
          <li
            class="govuk-tabs__list-item govuk-tabs__list-item--selected"
          >
            <a
              class="govuk-tabs__tab"
              href="#live"
            >
              Live
            </a>
          </li>
          <li
            class="govuk-tabs__list-item"
          >
            <a
              class="govuk-tabs__tab"
              href="#upcoming"
            >
              Upcoming
            </a>
          </li>
          <li
            class="govuk-tabs__list-item"
          >
            <a
              class="govuk-tabs__tab"
              href="#recently-closed"
            >
              Recently closed
            </a>
          </li>
        </ul>
        <div
          class="govuk-tabs__panel"
          id="live"
        >
          <table
            class="govuk-table"
          >
            <caption
              class="govuk-table__caption govuk-table__caption--l"
            >
              Live disruptions
            </caption>
            <thead
              class="govuk-table__head"
            >
              <tr
                class="govuk-table__row"
              >
                <th
                  class="govuk-table__header align-middle px-2"
                  scope="col"
                >
                  ID
                </th>
                <th
                  class="govuk-table__header align-middle px-2"
                  scope="col"
                >
                  Summary
                </th>
                <th
                  class="govuk-table__header align-middle px-2"
                  scope="col"
                >
                  Affected dates
                </th>
              </tr>
            </thead>
            <tbody
              class="govuk-table__body"
            />
          </table>
        </div>
        <div
          class="govuk-tabs__panel govuk-tabs__panel--hidden"
          id="upcoming"
        >
          <table
            class="govuk-table"
          >
            <caption
              class="govuk-table__caption govuk-table__caption--l"
            >
              Upcoming disruptions
            </caption>
            <thead
              class="govuk-table__head"
            >
              <tr
                class="govuk-table__row"
              >
                <th
                  class="govuk-table__header align-middle px-2"
                  scope="col"
                >
                  ID
                </th>
                <th
                  class="govuk-table__header align-middle px-2"
                  scope="col"
                >
                  Summary
                </th>
                <th
                  class="govuk-table__header align-middle px-2"
                  scope="col"
                >
                  Affected dates
                </th>
              </tr>
            </thead>
            <tbody
              class="govuk-table__body"
            />
          </table>
        </div>
        <div
          class="govuk-tabs__panel govuk-tabs__panel--hidden"
          id="recently-closed"
        >
          <table
            class="govuk-table"
          >
            <caption
              class="govuk-table__caption govuk-table__caption--l"
            >
              Closed disruptions
            </caption>
            <thead
              class="govuk-table__head"
            >
              <tr
                class="govuk-table__row"
              >
                <th
                  class="govuk-table__header align-middle px-2"
                  scope="col"
                >
                  ID
                </th>
                <th
                  class="govuk-table__header align-middle px-2"
                  scope="col"
                >
                  Summary
                </th>
                <th
                  class="govuk-table__header align-middle px-2"
                  scope="col"
                >
                  Affected dates
                </th>
              </tr>
            </thead>
            <tbody
              class="govuk-table__body"
            />
          </table>
        </div>
      </div>
      <a
        class="govuk-link"
        href="/view-all-disruptions"
      >
        <h2
          class="govuk-heading-s text-govBlue"
        >
          View all disruptions
        </h2>
      </a>
      <a
        class="govuk-link"
        href="/view-all-social-media"
      >
        <h2
          class="govuk-heading-s text-govBlue"
        >
          View all social media
        </h2>
      </a>
      <a
        class="govuk-link"
        href="/view-all-disruptions?draft=true"
      >
        <h2
          class="govuk-heading-s text-govBlue"
        >
          Draft disruptions
        </h2>
      </a>
      <a
        class="govuk-link"
        href="/view-all-templates"
      >
        <h2
          class="govuk-heading-s text-govBlue"
        >
          Templates
        </h2>
      </a>
      <a
        class="govuk-link"
        href="/view-all-roadworks"
      >
        <h2
          class="govuk-heading-s text-govBlue"
        >
          Create disruptions from roadworks in your area
        </h2>
      </a>
    </main>
    <div>
      <h2
        class="govuk-heading-s"
      >
        Help and Support
      </h2>
      <p
        class="govuk-body"
      >
        If you are having problems, please contact the Create Disruption Service via this link: 
        <a
          class="govuk-link govuk-!-font-size-19"
          href="/contact"
        >
          Contact us
        </a>
      </p>
    </div>
  </div>
  <footer
    class="govuk-footer"
    role="contentinfo"
  >
    <div
      class="govuk-width-container"
    >
      <div
        class="govuk-footer__meta"
      >
        <div
          class="govuk-footer__meta-item govuk-footer__meta-item--grow"
        >
          <h2
            class="govuk-visually-hidden"
          >
            Support links
          </h2>
          <ul
            class="govuk-footer__inline-list"
          >
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/contact"
              >
                Help
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/cookies"
              >
                Cookies
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/contact"
              >
                Contact
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/accessibility"
              >
                Accessibility
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="https://www.gov.uk/help/terms-conditions"
              >
                Terms and conditions
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/privacy"
              >
                Privacy
              </a>
            </li>
          </ul>
          <p
            class="govuk-footer__body"
          >
            Built by the 
            <a
              class="govuk-footer__link"
              href="https://www.gov.uk/government/organisations/department-for-transport"
              rel="noreferrer"
              target="_blank"
            >
              Department for Transport
            </a>
          </p>
          <svg
            aria-hidden="true"
            class="govuk-footer__licence-logo"
            focusable="false"
            height="17"
            viewBox="0 0 483.2 195.7"
            width="41"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145"
              fill="currentColor"
            />
          </svg>
          <span
            class="govuk-footer__licence-description"
          >
            All content is available under the 
            <a
              class="govuk-footer__link"
              href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
              rel="license"
            >
              Open Government Licence v3.0
            </a>
            , except where otherwise stated
          </span>
        </div>
        <div
          class="govuk-footer__meta-item"
        >
          <a
            class="govuk-footer__link govuk-footer__copyright-logo bg-govuk-crest"
            href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/"
          >
            Â© Crown copyright
          </a>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;

exports[`pages > dashboard > should render correctly for an operator user 1`] = `
<DocumentFragment>
  <div
    class="app-phase-banner__wrapper"
  >
    <div
      class="govuk-phase-banner max-w-[960px] w-[90%] border-0 mx-auto app-phase-banner app-width-container"
    >
      <p
        class="govuk-phase-banner__content"
      >
        <strong
          class="govuk-tag govuk-phase-banner__content__tag"
        >
          beta
        </strong>
        <span
          class="govuk-phase-banner__text"
        >
          This is a new service â€“ your 
          <a
            class="govuk-link"
            href="/feedback"
            id="feedback-link"
          >
            feedback
          </a>
           will help us to improve it.
        </span>
      </p>
    </div>
  </div>
  <div
    class="govuk-width-container"
  >
    <main
      class="govuk-main-wrapper"
      id="main-content"
    >
      <h1
        class="govuk-heading-xl"
      >
        Test Org disruptions data
      </h1>
      <a
        class="govuk-button govuk-button--start"
        data-module="govuk-button"
        draggable="false"
        href="/create-disruption/acde070d-8c4c-4f0d-9d8a-162843c10333"
        id="create-new-button"
        role="button"
      >
        Create new disruption
        <svg
          class="govuk-button__start-icon"
          focusable="false"
          height="19"
          role="presentation"
          viewBox="0 0 33 40"
          width="17.5"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h13l20 20-20 20H0l20-20z"
            fill="currentColor"
          />
        </svg>
      </a>
      <div
        class="govuk-tabs"
        data-module="govuk-tabs"
      >
        <h2
          class="govuk-tabs__title"
        >
          Disruptions
        </h2>
        <ul
          class="govuk-tabs__list"
        >
          <li
            class="govuk-tabs__list-item govuk-tabs__list-item--selected"
          >
            <a
              class="govuk-tabs__tab"
              href="#live"
            >
              Live
            </a>
          </li>
          <li
            class="govuk-tabs__list-item"
          >
            <a
              class="govuk-tabs__tab"
              href="#upcoming"
            >
              Upcoming
            </a>
          </li>
          <li
            class="govuk-tabs__list-item"
          >
            <a
              class="govuk-tabs__tab"
              href="#recently-closed"
            >
              Recently closed
            </a>
          </li>
        </ul>
        <div
          class="govuk-tabs__panel"
          id="live"
        >
          <table
            class="govuk-table"
          >
            <caption
              class="govuk-table__caption govuk-table__caption--l"
            >
              Live disruptions
            </caption>
            <thead
              class="govuk-table__head"
            >
              <tr
                class="govuk-table__row"
              >
                <th
                  class="govuk-table__header align-middle px-2"
                  scope="col"
                >
                  ID
                </th>
                <th
                  class="govuk-table__header align-middle px-2"
                  scope="col"
                >
                  Summary
                </th>
                <th
                  class="govuk-table__header align-middle px-2"
                  scope="col"
                >
                  Affected dates
                </th>
              </tr>
            </thead>
            <tbody
              class="govuk-table__body"
            />
          </table>
        </div>
        <div
          class="govuk-tabs__panel govuk-tabs__panel--hidden"
          id="upcoming"
        >
          <table
            class="govuk-table"
          >
            <caption
              class="govuk-table__caption govuk-table__caption--l"
            >
              Upcoming disruptions
            </caption>
            <thead
              class="govuk-table__head"
            >
              <tr
                class="govuk-table__row"
              >
                <th
                  class="govuk-table__header align-middle px-2"
                  scope="col"
                >
                  ID
                </th>
                <th
                  class="govuk-table__header align-middle px-2"
                  scope="col"
                >
                  Summary
                </th>
                <th
                  class="govuk-table__header align-middle px-2"
                  scope="col"
                >
                  Affected dates
                </th>
              </tr>
            </thead>
            <tbody
              class="govuk-table__body"
            />
          </table>
        </div>
        <div
          class="govuk-tabs__panel govuk-tabs__panel--hidden"
          id="recently-closed"
        >
          <table
            class="govuk-table"
          >
            <caption
              class="govuk-table__caption govuk-table__caption--l"
            >
              Closed disruptions
            </caption>
            <thead
              class="govuk-table__head"
            >
              <tr
                class="govuk-table__row"
              >
                <th
                  class="govuk-table__header align-middle px-2"
                  scope="col"
                >
                  ID
                </th>
                <th
                  class="govuk-table__header align-middle px-2"
                  scope="col"
                >
                  Summary
                </th>
                <th
                  class="govuk-table__header align-middle px-2"
                  scope="col"
                >
                  Affected dates
                </th>
              </tr>
            </thead>
            <tbody
              class="govuk-table__body"
            />
          </table>
        </div>
      </div>
      <a
        class="govuk-link"
        href="/view-all-disruptions"
      >
        <h2
          class="govuk-heading-s text-govBlue"
        >
          View all disruptions
        </h2>
      </a>
      <a
        class="govuk-link"
        href="/view-all-social-media"
      >
        <h2
          class="govuk-heading-s text-govBlue"
        >
          View all social media
        </h2>
      </a>
      <a
        class="govuk-link"
        href="/view-all-disruptions?draft=true"
      >
        <h2
          class="govuk-heading-s text-govBlue"
        >
          Draft disruptions
        </h2>
      </a>
      <a
        class="govuk-link"
        href="/view-all-roadworks"
      >
        <h2
          class="govuk-heading-s text-govBlue"
        >
          View roadworks in your area
        </h2>
      </a>
    </main>
    <div>
      <h2
        class="govuk-heading-s"
      >
        Help and Support
      </h2>
      <p
        class="govuk-body"
      >
        If you are having problems, please contact the Create Disruption Service via this link: 
        <a
          class="govuk-link govuk-!-font-size-19"
          href="/contact"
        >
          Contact us
        </a>
      </p>
    </div>
  </div>
  <footer
    class="govuk-footer"
    role="contentinfo"
  >
    <div
      class="govuk-width-container"
    >
      <div
        class="govuk-footer__meta"
      >
        <div
          class="govuk-footer__meta-item govuk-footer__meta-item--grow"
        >
          <h2
            class="govuk-visually-hidden"
          >
            Support links
          </h2>
          <ul
            class="govuk-footer__inline-list"
          >
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/contact"
              >
                Help
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/cookies"
              >
                Cookies
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/contact"
              >
                Contact
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/accessibility"
              >
                Accessibility
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="https://www.gov.uk/help/terms-conditions"
              >
                Terms and conditions
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/privacy"
              >
                Privacy
              </a>
            </li>
          </ul>
          <p
            class="govuk-footer__body"
          >
            Built by the 
            <a
              class="govuk-footer__link"
              href="https://www.gov.uk/government/organisations/department-for-transport"
              rel="noreferrer"
              target="_blank"
            >
              Department for Transport
            </a>
          </p>
          <svg
            aria-hidden="true"
            class="govuk-footer__licence-logo"
            focusable="false"
            height="17"
            viewBox="0 0 483.2 195.7"
            width="41"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145"
              fill="currentColor"
            />
          </svg>
          <span
            class="govuk-footer__licence-description"
          >
            All content is available under the 
            <a
              class="govuk-footer__link"
              href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
              rel="license"
            >
              Open Government Licence v3.0
            </a>
            , except where otherwise stated
          </span>
        </div>
        <div
          class="govuk-footer__meta-item"
        >
          <a
            class="govuk-footer__link govuk-footer__copyright-logo bg-govuk-crest"
            href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/"
          >
            Â© Crown copyright
          </a>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
